pkg/controller/job/job_controller_test.go

```diff
@@ -1904,7 +1909,19 @@ func TestSyncPastDeadlineJobFinished(t *testing.T) {
 			if err := sharedInformerFactory.Batch().V1().Jobs().Informer().GetIndexer().Add(job); err != nil {
 				t.Fatalf("Failed to insert job in index: %v", err)
 			}
+
 			var j *batch.Job
+			err = wait.PollImmediate(200*time.Microsecond, 3*time.Second, func() (done bool, err error) {
+				j, err = clientset.BatchV1().Jobs(metav1.NamespaceDefault).Get(ctx, job.GetName(), metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				return j.Status.StartTime != nil, nil
+			})
+			if err != nil {
+				t.Errorf("Job failed to ensure that start time was set: %v", err)
+			}
```

Comment:
We could check at this point that there is no deadline condition here

Reply:
@alculquicondor Sorry about the delay. I totally missed this comment/notification from you. Just pushed an updated version of the changes as suggested by you. PTAL when possible.